packages/ag-charts-community/src/chart/interaction/contextMenuRegistry.ts

```diff
@@ -22,22 +23,25 @@ type ContextEventProperties<K extends ContextType = ContextType> = {
 export type ContextType = keyof ContextTypeMap;
 export type ContextMenuEvent<K extends ContextType = ContextType> = ContextEventProperties<K> & ConsumableEvent;
 
-export type ContextMenuAction = {
+// Extract the TEvent types from the AgContextMenuOptions contract:
+type TEventOfContextMenuActions<K extends keyof Omit<AgContextMenuOptions, 'enabled'>> = Parameters<
+    NonNullable<AgContextMenuOptions[K]>[number]['action']
+>[0];
+export type ContextMenuCallback<K extends ContextType> = {
+    all: (params: TEventOfContextMenuActions<'extraActions'>) => void;
+    series: (params: TEventOfContextMenuActions<'extraNodeActions'>) => void;
+    legend: (params: TEventOfContextMenuActions<'extraLegendItemActions'>) => void;
+}[K];
```

Comment:
These types seem overly complex, could you simplify it to something like this to improve readability?
```
type ContextMenuActionMapping = {
    all: AgChartContextMenuEvent;
    legend: AgChartLegendContextMenuEvent;
    series: AgNodeContextMenuActionEvent;
};

export type ContextMenuAction<K extends ContextType> = {
    id?: string;
    label: string;
    type: K;
    action: (params: ContextMenuActionMapping[K]) => void;
};
```

If you wanted to keep the stricter references to `AgContextMenuOptions` then although more verbose, it is easier to follow:
```
type ContextMenuActionMapping = {
    all: Parameters<NonNullable<AgContextMenuOptions['extraActions']>[number]['action']>[0];
    legend: Parameters<NonNullable<AgContextMenuOptions['extraLegendItemActions']>[number]['action']>[0];
    series: Parameters<NonNullable<AgContextMenuOptions['extraNodeActions']>[number]['action']>[0];
};
```

Reply:
Yes, the intent was to use stricter references so that the TS compiler can check that we are running callbacks that matches the documentation. I don't have strong opinions whether with use the `ContextMenuActionMapping` or `TEventOfContextMenuActions` approach to achieve this, so happy to make this proposed change.